src/App.tsx

```diff
@@ -1,29 +1,288 @@
       <div className='App'>
-         <div className='score'>You have {count} antimatters.</div>
-         <div>You have {firstDimCount} First Dimension x2.0</div>
-         <button onClick={() => setFirstDimCount(firstDimCount + 1)}>Buy Dimension</button>
+         <div className='heading'>
+            <p className='centered highlight'>
+               You have {antimatter % 1 === 0 ? antimatter.toFixed(0) : antimatter.toFixed(1)}{' '}
+               antimatters.
+            </p>
+         </div>
+
+         <div className='gridContainer3Rows'>
+            <p className='centered'>{`The current clockspeed is ${clockSpeedRef.current.toFixed(
+               0
+            )} ms. Reduce the tickspeed by 11%.`}</p>
+            <div className='centered'>
+               <button
+                  className='btn'
+                  onClick={handleTickBtnClick}
+                  disabled={antimatter - tickspeedPrice <= 0}>
+                  Cost one time: {tickspeedPrice}{' '}
+               </button>
+               <button
+                  className='btn'
+                  onClick={handleBuyMaxClick}
+                  disabled={antimatter - tickspeedPrice <= 0}>
+                  Buy Max
+               </button>
+            </div>
+         </div>
+
+         <Dimension
+            antimatter={antimatter}
+            dimCount={firstDimCount}
+            setDimCount={setFirstDimCount}
+            setAntimatter={setAntimatter}
+            price={firstDimPrice}
+            setPrice={setFirstDimPrice}
+            factor={firstDimFactor}
+            setFactor={setFirstDimFactor}
+            dimFactorCount={firstDimFactorCount}
+            setDimFactorCount={setFirstDimFactorCount}>
+            {`First Dimension Cost: ${firstDimPrice}`}
+         </Dimension>
+
+         <Dimension
+            antimatter={antimatter}
+            dimCount={secondDimCount}
+            setDimCount={setSecondDimCount}
+            setAntimatter={setAntimatter}
+            price={secondDimPrice}
+            setPrice={setSecondDimPrice}
+            factor={secondDimFactor}
+            setFactor={setSecondDimFactor}
+            dimFactorCount={secondDimFactorCount}
+            setDimFactorCount={setSecondDimFactorCount}>
+            {`Second Dimension Cost: ${secondDimPrice}`}
+         </Dimension>
+         {/* the 3. dimension must be unlocked */}
+         {resetGameCounter > 2 && (
+            <Dimension
+               antimatter={antimatter}
+               dimCount={thirdDimCount}
+               setDimCount={setThirdDimCount}
+               setAntimatter={setAntimatter}
+               price={thirdDimPrice}
+               setPrice={setThirdDimPrice}
+               factor={thirdDimFactor}
+               setFactor={setThirdDimFactor}
+               dimFactorCount={thirdDimFactorCount}
+               setDimFactorCount={setThirdDimFactorCount}>
+               {`Third Dimension Cost: ${thirdDimPrice}`}
+            </Dimension>
```

Comment:
Nice work getting 3 dimensions to work!

You probably noticed that you have to copy a lot of code to add additional dimensions.  The next refactoring goal here would be to try to do something like this:
```
{dimensions.map((dim) =>(
  <Dimension
    dimCardinality={dim.cardinality} // which dim number is this?
    dimCount={dim.count}
    antimatter={antimatter}
    setDimCount={setDimCount} // when you call this function you'll pass the new count, plus the cardinality (which dim are we incrementing?)
    ...
))}
```
This will require you to refactor how you're saving state related to each dimension.  Probably you want to save an array of dimension objects, where each object has a set of fields containing the state related to that dimension.